src/types.jl

```diff
@@ -35,76 +31,73 @@ struct FontExtent{T}
+@inline topinkbound(ext::FontExtent) = hbearing_ori_to_top(ext)
 
 BroadcastStyle(::Type{<: FontExtent}) = Style{FontExtent}()
 BroadcastStyle(::Style{FontExtent}, x) = Style{FontExtent}()
 BroadcastStyle(x, ::Style{FontExtent}) = Style{FontExtent}()
 
-function broadcasted(op::Function, f::FontExtent, scaling::StaticVector)
-    return FontExtent(
+broadcasted(op::Function, f::FontExtent, scaling::StaticVector) =
+FontExtent(
         op.(f.vertical_bearing, scaling[1]),
         op.(f.horizontal_bearing, scaling[2]),
         op.(f.advance, scaling),
         op.(f.scale, scaling),
     )
-end
 
-function broadcasted(op::Function, f::FontExtent)
-    return FontExtent(
+broadcasted(op::Function, f::FontExtent) =
+    FontExtent(
         op.(f.vertical_bearing),
         op.(f.horizontal_bearing),
         op.(f.advance),
         op.(f.scale),
     )
-end
 
-function broadcasted(op::Function, ::Type{T}, f::FontExtent) where T
-    return FontExtent(
+broadcasted(op::Function, ::Type{T}, f::FontExtent) where T =
+    FontExtent(
         map(x-> op(T, x), f.vertical_bearing),
         map(x-> op(T, x), f.horizontal_bearing),
         map(x-> op(T, x), f.advance),
         map(x-> op(T, x), f.scale),
     )
-end
 
-function FontExtent(fontmetric::FreeType.FT_Glyph_Metrics, scale::T = 64.0) where T <: AbstractFloat
-    return FontExtent(
+FontExtent(fontmetric::FreeType.FT_Glyph_Metrics, scale::T = 64.0) where T <: AbstractFloat =
+    FontExtent(
         Vec{2, T}(fontmetric.vertBearingX, fontmetric.vertBearingY) ./ scale,
         Vec{2, T}(fontmetric.horiBearingX, fontmetric.horiBearingY) ./ scale,
         Vec{2, T}(fontmetric.horiAdvance, fontmetric.vertAdvance) ./ scale,
         Vec{2, T}(fontmetric.width, fontmetric.height) ./ scale
     )
-end
 
-function ==(x::FontExtent, y::FontExtent)
-    return (x.vertical_bearing == y.vertical_bearing &&
-            x.horizontal_bearing == y.horizontal_bearing &&
-            x.advance == y.advance &&
-            x.scale == y.scale)
-end
+==(x::FontExtent, y::FontExtent) = (
+    x.vertical_bearing == y.vertical_bearing &&
+    x.horizontal_bearing == y.horizontal_bearing &&
+    x.advance == y.advance &&
+    x.scale == y.scale
+)
 
-function FontExtent(fontmetric::FreeType.FT_Glyph_Metrics, scale::Integer)
-    return FontExtent(
+FontExtent(fontmetric::FreeType.FT_Glyph_Metrics, scale::Integer) =
+    FontExtent(
         div.(Vec{2, Int}(fontmetric.vertBearingX, fontmetric.vertBearingY), scale),
         div.(Vec{2, Int}(fontmetric.horiBearingX, fontmetric.horiBearingY), scale),
         div.(Vec{2, Int}(fontmetric.horiAdvance, fontmetric.vertAdvance), scale),
         div.(Vec{2, Int}(fontmetric.width, fontmetric.height), scale)
     )
-end
 
-function bearing(extent::FontExtent)
-    return Vec2f(extent.horizontal_bearing[1],
-                  -(extent.scale[2] - extent.horizontal_bearing[2]))
-end
+bearing(extent::FontExtent{T}) where T = 
```

Comment:
I think this is just a convention where you start the bearing, and this turned out as the most useful convention for the textlayouting work in Makie.
Do you have a good reason to change this and turn this into a breaking release?

Reply:
Ok, maybe I didn't understand correctly what was computed. It was initially for drawing bounding boxes in https://github.com/t-bltg/FreeTypeAbstraction.jl/blob/ab35f9ee3b8d3dd950ad51fa166841e47f156814/src/rendering.jl#L189, and followed the convention of https://github.com/t-bltg/FreeTypeAbstraction.jl/blob/ab35f9ee3b8d3dd950ad51fa166841e47f156814/src/rendering.jl#L138 (`x` pointing to the east and `y` to the south, where in FreeType, `y` points to the north), and worked nicely with the example codes given in https://github.com/JuliaGraphics/FreeTypeAbstraction.jl/pull/71#issue-1181220071, but if this gonna break `Makie`, I'm going to change the algorithm.

Reply:
I honestly think that the previous drawing code, using the bearing fix should work using `boundingbox`:
```julia
rect = boundingbox(metric)
(mc, mr), (Mc, Mr) = extrema(rect)
r1, r2 = clamp(py + mr, 1, imgh), clamp(py + Mr, 1, imgh)
c1, c2 = clamp(px + mc, 1, imgw), clamp(px + Mc, 1, imgw)
```

But since I had an alternative drawing algorithm commented out, which works regardless of the way bearing is computed, I'm ok to revert the changes:
```julia
r1, r2 = clamp(oy + row_lo, 1, imgh), clamp(oy + row_hi, 1, imgh)
c1, c2 = clamp(ox + col_lo, 1, imgw), clamp(ox + col_hi, 1, imgw)
```